### PR TITLE
Authorization checker

### PR DESCRIPTION
**Pull request by cocoweb94**
## Description

Add adapter authorizationChecker and establishment of authorizations on catalog and form product
## Steps to Test this Fix

1. change authorization on one profile 
2. connect to BO and see all action is congruent of config in BO
## Forge Ticket (optional)

# BOOM-393

### CODE REVIEW

File: app/config/config.yml [hunk 1]
```diff
@@ -8,6 +8,7 @@ imports:
 parameters:
     locale: en
     AdapterSecurityAdminClass: PrestaShop\PrestaShop\Adapter\Security\Admin
+    authorization_employee: ['view','add','edit','delete']
```
Comment (mickaelandrieu) [inline app/config/config.yml]: [YAGNI](https://fr.wikipedia.org/wiki/YAGNI) I'd like to see theses authorizations to be part of AuthorizationChecker class.

File: src/Adapter/Security/AuthorizationChecker.php [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+use Symfony\Component\HttpKernel\KernelInterface;
+
+/**
+ * Admin Middleware security
+ */
+class AuthorizationChecker implements AuthorizationCheckerInterface
+{
+    private $legacyContext;
+    private $container;
+
```
Comment (mickaelandrieu) [inline src/Adapter/Security/AuthorizationChecker.php]:
```php
const RIGHTS = ['add', 'delete', 'edit','view'];
```

File: src/Adapter/Security/AuthorizationChecker.php [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    /**
+     * Constructor.
+     *
+     * @param LegacyContext $context
+     */
+    public function __construct(LegacyContext $context, KernelInterface $kernel)
```
Comment (mickaelandrieu) [inline src/Adapter/Security/AuthorizationChecker.php]: 2 things here:
* You don't need Kernel dependency, but container. If you use a constant, you don't need it, though...
* Avoid use of LegacyContext when you can, you can directly depend on Employee here

File: src/Adapter/Security/AuthorizationChecker.php [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\Security;
+
+use Symfony\Component\Security\Core\Authorization\AuthorizationCheckerInterface;
+use PrestaShop\PrestaShop\Adapter\LegacyContext;
+use Symfony\Component\HttpKernel\KernelInterface;
+
+/**
+ * Admin Middleware security
```
Comment (mickaelandrieu) [inline src/Adapter/Security/AuthorizationChecker.php]: Be careful, it's not a middleware the way you use it :)

**Comment by mickaelandrieu** [comment]
At least I have a question: why not use a dedicated Voter and let the framework check for authorizations itself ? http://symfony.com/doc/current/cookbook/security/voters.html

**Comment by Shudrum** [comment]
The idea is to use the same interface that the security system of Symfony and try to use it well to easily switch to the Symfony class when needed.

**Comment by cocoweb94** [comment]
The VoterInterface have working interesting but it seems to me that she can check permission on object and not on route. However it seems to me that all permissions in gestion of rights in BO are do by route and not by object.

**Comment by mickaelandrieu** [comment]
@cocoweb94 http://symfony.com/doc/current/components/security/authorization.html#the-access-listener

if you are correctly set up and registered the voter, it should be applied to any Symfony route of your application.

**Comment by maximebiloe** [comment]
I think it can interest you @jtabet :wink:

**Comment by maximebiloe** [comment]
All the right management system has been reworked here:https://github.com/PrestaShop/PrestaShop/pull/5390
So this PR is no longer needed.

Thanks

**Pull request closed without merging**